### PR TITLE
DDF-04518 Cleared query attribute value when switching between attributes of different types

### PR DESCRIPTION
**Pull request by hayleynorton**
#### What does this PR do?
When updating the filter, the type of the previously selected attribute is compared to the type of the currently selected attribute. If they're not the same type, the value is cleared to avoid unexpected/undesired values. 
#### Who is reviewing it? 
@andrewzimmer @Bdthomson 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining 
@djblue 
@millerw8 
#### How should this be tested?
- Switch to `anyGeo` and draw a shape
- Switch to `anyText` and verify that the value is empty
- Fill in a value and switch to a Boolean attribute
- Verify that the value is empty
- Switch to any other String attribute and enter a number
- Switch to a Date attribute and verify that the value is empty
- Choose a date and switch to another Date attribute
- Verify that the same date remains
- Switch between other attributes of the same type and make sure that the values remain (with the exception of different shapes: polygon to bounding box, bounding box to point radius, etc)
#### What are the relevant tickets?
Fixes: #4518 
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js [hunk 1]
```diff
@@ -222,10 +222,17 @@ module.exports = Marionette.LayoutView.extend({
     this.filterAttribute.currentView.model.set('value', [attribute])
   },
   handleAttributeUpdate: function() {
+    const previousAttributeType =
+      metacardDefinitions.metacardTypes[this.model.get('type')].type
     this.model.set(
       'type',
       this.filterAttribute.currentView.model.get('value')[0]
     )
+    const currentAttributeType =
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]:
```suggestion
    handleAttributeUpdate: function() {
    const previousAttributeType = metacardDefinitions.metacardTypes[this.model.get('type')].type
    const nextAttributeType = this.filterAttribute.currentView.model.get('value')[0]
    this.model.set({
        type: nextAttributeType,
        ...(previousAttributeType !== nextAttributeType) && { value: ['']}
    });
    }
```

Comment (hayleynorton) [reply ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]: I'm not sure that will compare the types I had in mind. Here, `previousAttributeType` will be something like `STRING` or `LOCATION`, but `nextAttributeType` will be something like `anyText` or `anyGeo`

**Comment by andrewzimmer** [comment]
Hero Successful 🎉 

- verified the value is cleared when switching types
- verified value remained when switching between same type (text, date, boolean)

**Comment by andrewzimmer** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js (hunk 1 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]: Ah, I see, the declaration of the nextAttributeType should be:
```
 const nextAttributeType = metacardDefinitions.metacardTypes[this.filterAttribute.currentView.model.get('value')[0]].type
```

**Comment by willwill96** [comment]
build now

**Pull request merged**